opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -151,6 +162,13 @@ impl<R: RuntimeChannel> LogProcessor for BatchLogProcessor<R> {
             .map_err(|err| LogError::Other(err.into()))
             .and_then(std::convert::identity)
     }
+
+    fn set_resource(&self, resource: &Resource) {
+        let resource = Arc::new(resource.clone());
```

Comment:
could we clone the resource, and then hand it over to the BatchMessage::SetResource? Is Arc needed? The processor need not keep ownership, so it should be okay to move ownership to BatchMessage?

Reply:
Yes Arc is needed for batch-processor, to share the resource across separate thread. I believe it should be fine to have one-time overhead during startup.